lib/fluent/plugin/filter_masking.rb

```diff
@@ -11,18 +11,37 @@ def strToHash(str)
         eval(str)
       end
 
+      def myDig(input, path)
+        curr = input
+        for segment in path do
+          if curr != nil && curr.respond_to?(:dig)
+            if curr[segment] == nil
+              curr = curr[segment.to_s]
```

Comment:
if `curr[segment] == nil` then woudn't `curr[segment.to_s]` also equal `nil`?

Reply:
No, a symbol and a string are different hash keys, that's what made it work on dev.

Reply:
then i would add a comment for this line and the next one like this: 
`curr = curr[segment.to_s]    //string type` 
`curr = curr[segment]    //symbol type` 

wdyt?